docs/apworld_dev_faq.md

```diff
@@ -43,3 +43,25 @@ A faster alternative to the `for` loop would be to use a [list comprehension](ht
 ```py
 item_pool += [self.create_filler() for _ in range(total_locations - len(item_pool))]
 ```
+
+---
+
+### I learned about indirect conditions in the world API document, but I want to know more. What are they and why are they necessary?
+
+The world API document mentions indirect conditions and **when** you should use them, but not *how* they work and *why* they are necessary. This is because the explanation is quite complicated.
+
+Region sweep (the algorithm that determines which regions are reachable) is a Breadth First Search of the region graph from the Menu region, checking entrances one by one and adding newly reached nodes (regions) and their entrances to the queue until there is nothing more to check.
+
+For performance reasons AP only checks every entrance once. However, if entrance access conditions depend on regions, then it is possible for this to happen:
```

Comment:
```suggestion
For performance reasons, AP only checks every entrance once. However, if entrance access conditions depend on regions, then it is possible for this to happen:
```

Reply:
is that comma proper grammar? imo it reads better without it

Reply:
It is, although looking into a few different guidebooks, it seems there's disagreement on whether short prepositional phrases ("short" being two, three, four, or five words, depending on the guidebook) actually need a comma. I see what you mean though, feel free to ignore this comma